refinery/tool_manager/models.py

```diff
@@ -449,40 +456,127 @@ class Meta:
-                        file_mapping[self.REFINERY_FILE_UUID]
-                    ),
-                    'src': 'hda'
-                }
-                for file_mapping in self.get_file_relationships_galaxy()
-            ]
-        }
+        flattened_nesting = self._flatten_file_relationships_nesting()
+        for index, file_relationship in enumerate(flattened_nesting):
+            if isinstance(file_relationship, list):
+                nesting_string += "{}:".format(WorkflowTool.LIST)
+            if isinstance(file_relationship, tuple):
+                nesting_string += "{}:".format(WorkflowTool.PAIRED)
+        return nesting_string[:-1]
+
+    @property
+    def galaxy_history_id(self):
+        return self.get_galaxy_dict()[self.GALAXY_IMPORT_HISTORY_DICT]["id"]
+
+    def _associate_collection_elements(self, galaxy_element_data):
+        """
+        Handles the association of Galaxy objects with their parent elements
+        :param galaxy_element_data: list of bioblend objects to associate
+        with each other
+        :return: <CollectionDescription>
+        """
+        if not len(galaxy_element_data) == 1:
+            for index, galaxy_element in enumerate(galaxy_element_data):
+                if type(galaxy_element) == CollectionDescription:
+                    for prev_galaxy_element in galaxy_element_data[:index]:
+                        if (type(prev_galaxy_element) ==
+                                HistoryDatasetElement or
+                                type(prev_galaxy_element) ==
+                                CollectionElement):
+                            galaxy_element.add(prev_galaxy_element)
+                            galaxy_element_data.remove(prev_galaxy_element)
+                            return self._associate_collection_elements(
+                                galaxy_element_data
+                            )
+                if type(galaxy_element) == CollectionElement:
+                    for prev_galaxy_element in galaxy_element_data[:index]:
+                        if type(prev_galaxy_element) == HistoryDatasetElement:
+                            galaxy_element.add(prev_galaxy_element)
+                            galaxy_element_data.remove(prev_galaxy_element)
+                            return self._associate_collection_elements(
+                                galaxy_element_data
+                            )
+                        if (type(prev_galaxy_element) == CollectionElement and
+                                prev_galaxy_element.type !=
+                                galaxy_element.type):
+                            galaxy_element.add(prev_galaxy_element)
+                            galaxy_element_data.remove(prev_galaxy_element)
+                            return self._associate_collection_elements(
+                                galaxy_element_data
+                            )
+        else:
+            assert len(galaxy_element_data) == 1
```

Comment:
These asserts probably aren't necessary, but may be good to keep for now while this feature is in a nubile stage. 

- [ ] Add comments
- [ ] Add messages to be included in AssertionErrors